local-interchaintest/examples/ibc_transfer_juno_ntrn.rs

```diff
@@ -0,0 +1,305 @@
+};
+use localic_std::modules::{
+    bank,
+    cosmwasm::{contract_execute, contract_instantiate},
+};
+use localic_utils::{
+    ConfigChainBuilder, TestContextBuilder, DEFAULT_KEY, JUNO_CHAIN_ADMIN_ADDR, JUNO_CHAIN_NAME,
+    LOCAL_IC_API_URL, NEUTRON_CHAIN_NAME,
+};
+use log::info;
+
+use valence_generic_ibc_transfer_service::msg::{
+    ActionMsgs, IbcTransferAmount, ServiceConfig, ServiceConfigUpdate,
+};
+use valence_service_utils::{denoms::UncheckedDenom, ServiceAccountType};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    env_logger::init();
+
+    let mut test_ctx = TestContextBuilder::default()
+        .with_unwrap_raw_logs(true)
+        .with_api_url(LOCAL_IC_API_URL)
+        .with_artifacts_dir(VALENCE_ARTIFACTS_PATH)
+        .with_chain(ConfigChainBuilder::default_neutron().build()?)
+        .with_chain(ConfigChainBuilder::default_juno().build()?)
+        .with_log_file_path(LOGS_FILE_PATH)
+        .with_transfer_channels(NEUTRON_CHAIN_NAME, JUNO_CHAIN_NAME)
+        .build()?;
+
+    let neutron_on_juno_denom = test_ctx
+        .get_ibc_denom()
+        .base_denom(NTRN_DENOM.to_owned())
+        .src(NEUTRON_CHAIN_NAME)
+        .dest(JUNO_CHAIN_NAME)
+        .get();
+
+    // Let's upload the base account contract to Juno
+    let current_dir = env::current_dir()?;
+    let base_account_contract_path = format!(
+        "{}/artifacts/valence_base_account.wasm",
+        current_dir.display()
+    );
+
+    let mut uploader = test_ctx.build_tx_upload_contracts();
+    uploader
+        .with_chain_name(JUNO_CHAIN_NAME)
+        .send_single_contract(&base_account_contract_path)?;
+
+    // Get the code id
+    let code_id_base_account = *test_ctx
+        .get_chain(JUNO_CHAIN_NAME)
+        .contract_codes
+        .get("valence_base_account")
+        .unwrap();
+
+    // Create 1 base accounts on Juno, to be the input account for the IBC transfer service
+    let base_accounts = create_base_accounts(
+        &mut test_ctx,
+        DEFAULT_KEY,
+        JUNO_CHAIN_NAME,
+        code_id_base_account,
+        JUNO_CHAIN_ADMIN_ADDR.to_string(),
+        vec![],
+        1,
+    );
+    let input_account = base_accounts[0].clone();
+    info!("Input account: {:?}", input_account);
+
+    // Send NTRN tokens to the input account on Juno
+    test_ctx
+        .build_tx_transfer()
+        .with_chain_name(NEUTRON_CHAIN_NAME)
+        .with_amount(1_000_000_000_000u128)
+        .with_recipient(&input_account)
+        .with_denom(NTRN_DENOM)
+        .send()
+        .unwrap();
+    std::thread::sleep(std::time::Duration::from_secs(3));
+
+    let start_input_balance = bank::get_balance(
+        test_ctx
+            .get_request_builder()
+            .get_request_builder(JUNO_CHAIN_NAME),
+        &input_account,
+    )
+    .iter()
+    .find(|bal| bal.denom == neutron_on_juno_denom)
+    .map_or(0, |bal| bal.amount.u128());
+    info!("Start input balance: {:?}", start_input_balance);
```

Comment:
nit: might be worth checking this right before the transfer execution as IBC transfers might take a bit longer (they depend on the relayer). This way you won't get 0 sometimes here, or just sleep for a bit longer (relayer is sometimes very slow on CI)

Reply:
So you mean the sleep just before that should be a bit longer, right ?